solidity/contracts/BancorNetwork.sol

```diff
@@ -636,70 +642,58 @@ contract BancorNetwork is IBancorNetwork, TokenHolder, ContractRegistryClient, R
 
     bytes4 private constant GET_RETURN_FUNC_SELECTOR = bytes4(keccak256("getReturn(address,address,uint256)"));
 
-    // using assembly code since older converter versions have different return values
-    function getReturn(address _dest, address _sourceToken, address _targetToken, uint256 _amount) internal view returns (uint256, uint256) {
-        uint256[2] memory ret;
+    // using a static call to get the return from older converters
+    function getReturn(IConverter _dest, IERC20Token _sourceToken, IERC20Token _targetToken, uint256 _amount) internal view returns (uint256, uint256) {
         bytes memory data = abi.encodeWithSelector(GET_RETURN_FUNC_SELECTOR, _sourceToken, _targetToken, _amount);
+        (bool success, bytes memory returnData) = address(_dest).staticcall(data);
+
+        if (success) {
+            if (success && returnData.length == 64) {
+                return abi.decode(returnData, (uint256,uint256));
+            }
 
-        assembly {
-            let success := staticcall(
-                gas,           // gas remaining
-                _dest,         // destination address
-                add(data, 32), // input buffer (starts after the first 32 bytes in the `data` array)
-                mload(data),   // input length (loaded from the first 32 bytes in the `data` array)
-                ret,           // output buffer
-                64             // output length
-            )
-            if iszero(success) {
-                revert(0, 0)
+            if (success && returnData.length == 32) {
+                return (abi.decode(returnData, (uint256)), 0);
             }
         }
 
-        return (ret[0], ret[1]);
+        return (0, 0);
     }
 
     bytes4 private constant IS_V28_OR_HIGHER_FUNC_SELECTOR = bytes4(keccak256("isV28OrHigher()"));
 
-    // using assembly code to identify converter version
+    // using a static call to identify converter version
     // can't rely on the version number since the function had a different signature in older converters
     function isV28OrHigherConverter(IConverter _converter) internal view returns (bool) {
-        bool success;
-        uint256[1] memory ret;
         bytes memory data = abi.encodeWithSelector(IS_V28_OR_HIGHER_FUNC_SELECTOR);
+        (bool success, bytes memory returnData) = address(_converter).staticcall{ gas: 4000 }(data);

```

Comment:
It was `5000` before. Was there any specific reason you've changed it to `4000` (not that it'd matter that much)?

Reply:
reduce gas costs by another 1K and also storage update is 5K gas (not that it matters for a staticcall)

Reply:
Any chance that Solidity-level `staticcall` uses only the exact amount of gas required, and not the entire amount of gas remaining (in which case, we can remove this hard-coded value)?

Reply:
I highly doubt it, as it's exactly like calling a view function from within a tx.